File: python/1734_解码异或后的排列_medium.py
class Solution:
    def decode(self, encoded: List[int]) -> List[int]:
        n, total = len(encoded), 0
        ans = [0] * (n+1)
        for i in range(1, n+2):
            total ^= i
        for i in range(1, n, 2):
            ans[0] ^= encoded[i]
        ans[0] ^= total
        for i in range(1, n+1):
            ans[i] = ans[i-1] ^ encoded[i-1]
        return ans

File: cpp/1734_解码异或后的排列_medium.cpp
class Solution {
public:
    vector<int> decode(vector<int>& encoded) {
        int n = encoded.size();
        vector<int> ans(n+1);
        int total = 0;
        for (int i=1; i<=n+1;i++)
            total ^= i;
        for (int i=1; i<n; i+=2)
            ans[0] ^= encoded[i];
        ans[0] ^= total;
        for (int i=1; i<=n; i++)
            ans[i] = ans[i-1] ^ encoded[i-1];
        return ans;
    }
};